sympy/parsing/mathematica.py

```diff
@@ -42,6 +42,22 @@ def parse(s):
         (r"\A(-? *[\d\.]+)([a-zA-Z].*)\Z",  # Implied multiplication - 2a
         lambda m: parse(m.group(1)) + "*" + parse(m.group(2))),
 
+        (r"\A([^\+\-]*[\w\.]) ([^\+\-]+)\Z",  # Space multiplication - x sin[x]
```

Comment:
Ah yes indeed! It's a bit trivial and I must have overlooked strings like that; working on it. Should I break the it into two commits, one for the issue and the other for space multiplication, so that issue gets its own commit and reverting changes or working for space multiplication at later stage should be convenient?